### ISSUE TITLE
`sourcemaps.deleteSourcemapsAfterUpload` breaks Vercel deployment

### ISSUE THREAD

**Issue by ocavue** [getsentry/sentry#75216]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

I'm using Vercel to deploy my Next.js project. When I enable `sourcemaps: { deleteSourcemapsAfterUpload: true }` in `next.config.js`, I get the following error from Vercel:

```

Route (pages)                              Size     First Load JS
┌ ○ /                                      8.71 kB         163 kB
├   └ css/88d021db95f13dbf.css             2.02 kB
├   /_app                                  0 B             154 kB
├ ○ /404                                   438 B           155 kB
├ ƒ /api/sentry-example-api                0 B             154 kB
├ ƒ /api/test1                             0 B             154 kB
├ ƒ /api/test2                             0 B             154 kB
├ ƒ /api/test3                             0 B             154 kB
├ ƒ /api/test4                             0 B             154 kB
├ ○ /client/test1                          429 B           155 kB
├ ○ /client/test2                          456 B           155 kB
├ ○ /client/test3                          445 B           155 kB
├ ○ /client/test4                          467 B           155 kB
├ ○ /sentry-example-page                   1.86 kB         156 kB
├ ƒ /ssr/test1                             431 B           155 kB
├ ƒ /ssr/test2                             429 B           155 kB
├ ƒ /ssr/test3                             430 B           155 kB
└ ƒ /ssr/test4                             432 B           155 kB
+ First Load JS shared by all              155 kB
  ├ chunks/framework-8722668362422965.js   45.3 kB
  ├ chunks/main-4d4ce5325c2179ff.js        32.2 kB
  ├ chunks/pages/_app-c18a80145f513654.js  76 kB
  └ other shared chunks (total)            1.66 kB
○  (Static)   prerendered as static content
ƒ  (Dynamic)  server-rendered on demand

Error: ENOENT: no such file or directory, lstat '/vercel/path0/.next/server/chunks/195.js.map'
```

I've created a minimal repo to reproduce this issue: https://github.com/ocavue/nextjs-sentry-vercel-deploy-issue

You can see from the GitHub webpage that the Vercel deployment failed after I enable `deleteSourcemapsAfterUpload`. 

![Image](https://github.com/user-attachments/assets/76d414fa-d96e-4de6-9f4d-fb93e796de9f)

### Expected Result

No error.

### Actual Result

Vercel deployment fails, as shown above. 

### Product Area

Unknown

### Version

8.20.0

**Comment by ocavue** [comment]
It seems that when `deleteSourcemapsAfterUpload` is enable, all `.js.map` files under `.next/` are deleted. ([source code]( https://github.com/getsentry/sentry-javascript/blob/f61e729423/packages/nextjs/src/config/webpackPluginOptions.ts#L77-L83 ))

And based on the error message above, it seems that Vercel does need `.next/server/chunks/195.js.map`.

**Comment by ocavue** [comment]
Cloudflare R2 is AWS S3 compatible, and `reflect-assets` is a relatively simple wrapper around Cloudflare R2. So S3's API is the best reference here.

---

I just checked again S3's API "[User-defined object metadata](https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html)" section. Here is what they design. 

You can add `x-amz-meta-` prefix to the HTTP header to storage custom metadata.

If your HTTP header value is just an ASCII string, you can just provide it as it's in the HTTP request, and the HTTP response will reply to it as it's.

```
PUT /Key HTTP/1.1
Host: DOC-EXAMPLE-BUCKET1.s3.amazonaws.com
x-amz-meta-ascii: AMAZONS3

HEAD /Key HTTP/1.1
Host: DOC-EXAMPLE-BUCKET1.s3.amazonaws.com
x-amz-meta-ascii: AMAZONS3
```

If your HTTP header value contains non-ASCII string, you should use _RFC 2047 encoding_ to encoding it. Here is what _RFC 2047 encoding_ looks like:

```
HEAD /Key HTTP/1.1
Host: DOC-EXAMPLE-BUCKET1.s3.amazonaws.com
x-amz-meta-nonascii: =?UTF-8?B?w4PChE3Dg8KEWsODwpXDg8KRIFMz?=
```

- `=?UTF-8?B?`: Prefix, mark it using base64 encoding UTF-8 string
- `w4PChE3Dg8KEWsODwpXDg8KRIFMz`: The base64 string
- `?=`: Suffix

In short, AWS S3 also uses base64 for custom metadata. I can use _RFC 2047_ (instead of base64 directly) so that it's closer to S3's design, but they don't have that much different IMO.

---

Another path is that we don't allow custom metadata at all. We can define a fixed set of metadata like `filename`, `content-type`, etc. We don't need base64 anymore because we're not accepting an ambiguous JSON object. The downside is we sacrifice flexibility. S3 API also has a set of fixed list of metadata they accept.

---

Which path do you prefer?
